frontend/pkg/frontend/middleware_tracing_test.go

```diff
@@ -72,13 +64,8 @@ func TestMiddlewareTracing(t *testing.T) {
 	} {
 		t.Run(tc.name, func(t *testing.T) {
 			// Setup the testing tracer.
-			inMemoryExporter := tracetest.NewInMemoryExporter()
```

Comment:
fyi @frzifus I've reworked a bit the test framework for tracing. let me know what you think.